src/test/function_test/test_bulk_load.cpp

```diff
@@ -0,0 +1,254 @@
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    void operate_data(bulk_load_test::operation op, const std::string &value, int count)
+    {
+        for (int i = 0; i < count; ++i) {
+            std::string hash_key = HASHKEY_PREFIX + boost::lexical_cast<std::string>(i);
+            std::string sort_key = SORTKEY_PREFIX + boost::lexical_cast<std::string>(i);
+            switch (op) {
+            case bulk_load_test::operation::GET: {
+                std::string act_value;
+                int ret = pg_client->get(hash_key, sort_key, act_value);
+                ASSERT_EQ(ret, 0);
+                ASSERT_EQ(act_value, value);
+            } break;
+            case bulk_load_test::operation::DEL: {
+                int ret = pg_client->del(hash_key, sort_key);
+                ASSERT_EQ(0, ret);
+            } break;
+            case bulk_load_test::operation::SET: {
+                int ret = pg_client->set(hash_key, sort_key, value);
+                ASSERT_EQ(0, ret);
+            } break;
+            case bulk_load_test::operation::NO_VALUE: {
+                std::string act_value;
+                int ret = pg_client->get(hash_key, sort_key, act_value);
+                // -1001 means value not found
+                ASSERT_EQ(ret, -1001);
+            } break;
+            default:
+                break;
+            }
+        }
+    }
+
+    const std::string LOCAL_ROOT = "bulk_load_root";
+    const std::string CLUSTER = "cluster";
+    const std::string APP_NAME = "temp";
+    const std::string PROVIDER = "local_service";
+
+    const std::string HASHKEY_PREFIX = "hash";
+    const std::string SORTKEY_PREFIX = "sort";
+    const std::string VALUE = "newValue";
+    const int32_t COUNT = 1000;
+};
+
+///
+/// case1: lack of `bulk_load_info` file
+/// case2: `bulk_load_info` file inconsistent with app_info
+///
+TEST_F(bulk_load_test, bulk_load_test_failed)
+{
+    // bulk load failed because `bulk_load_info` file is missing
+    remove_file(bulk_load_local_root + "/" + CLUSTER + "/" + APP_NAME + "/bulk_load_info");
+    ASSERT_EQ(start_bulk_load(), ERR_OBJECT_NOT_FOUND);
+
+    // bulk load failed because `bulk_load_info` file inconsistent with current app_info
+    replace_bulk_load_info();
+    ASSERT_EQ(start_bulk_load(), ERR_INCONSISTENT_STATE);
+}
+
+///
+/// case1: lack of `bulk_load_metadata` file
+/// case2: bulk load succeed with data verfied
+/// case3: bulk load data consistent:
+///     - old data will be overrided by bulk load data
+///     - get/set/del succeed after bulk load
+///
+TEST_F(bulk_load_test, bulk_load_tests)
+{
+    // bulk load failed because partition[0] `bulk_load_metadata` file is missing
+    remove_file(bulk_load_local_root + "/" + CLUSTER + "/" + APP_NAME + "/0/bulk_load_metadata");
+    ASSERT_EQ(start_bulk_load(), ERR_OK);
+    // bulk load will get FAILED
```

Comment:
How do you know bulk load will get FAILED indeed?

Reply:
I update the function `wait_bulk_load_finish`, it will return last bulk load status.